platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpoint.java

```diff
@@ -159,73 +155,65 @@ public Response create(MultipartBody multipartBody, @Context UriInfo requestUriI
    * @param attachment the attachment to copy and extract.
    * @return The file of the copied attachment.
    */
+  @Nullable
   private File createFileFromAttachement(Attachment attachment) {
-    InputStream inputStream = null;
-    String filename = null;
-    File newFile = null;
-    if (attachment.getContentDisposition() != null) {
-      filename =
-          attachment
-              .getContentDisposition()
-              .getParameter(FILENAME_CONTENT_DISPOSITION_PARAMETER_NAME);
-    }
-
-    if (StringUtils.isEmpty(filename)) {
-      LOGGER.debug("Filename not found, using default.");
-      filename = DEFAULT_FILE_NAME;
-    } else {
-      filename = FilenameUtils.getName(filename);
-      LOGGER.debug("Filename: {}", filename);
-    }
+    try (InputStream inputStream = attachment.getDataHandler().getInputStream()) {
+      if (inputStream == null) {
+        LOGGER.debug("No file attachment found");
+        return null;
+      }
 
-    try {
-      inputStream = attachment.getDataHandler().getInputStream();
-      if (inputStream != null && inputStream.available() == 0) {
+      if (inputStream.available() == 0) {
         inputStream.reset();
       }
+
+      String filename = getFileName(attachment.getContentDisposition());
+
+      if (!filename.endsWith(JAR_EXT) && !filename.endsWith(KAR_EXT)) {
+        LOGGER.debug("Wrong file type: {}", FilenameUtils.getExtension(filename));
+        Response.ResponseBuilder responseBuilder = Response.serverError();
+        responseBuilder.status(HttpStatus.UNSUPPORTED_MEDIA_TYPE_415);
+        return null;
+      }
+
+      File uploadDir = new File(defaultFileLocation);
+      if (!uploadDir.exists() && uploadDir.mkdirs()) {
+        LOGGER.info("Unable to make directory {}", uploadDir.getAbsolutePath());
```

Comment:
❓ Shouldn't this message be printed if `uploadDir.mkdirs()` returns false? It seems like right now it's printing `Unable to make directory` even if it creates it successfully.

Reply:
Yeah I noticed that. Probably an existing bug. Worth a patch IMO if no one else disagrees.

Reply:
Yeah I think it's worth fixing here